internal/extproc/watcher.go

```diff
@@ -77,6 +79,9 @@ func (cw *configWatcher) loadConfig(ctx context.Context) error {
 		return err
 	}
 
+	cw.mu.Lock()
+	defer cw.mu.Unlock()
```

Comment:
i believe you shouldn't need the change like this whose purpose is only for testing. The code path itself here  is not concurrently called, so this mutex shouldn't exist.